### PR TITLE
WIP: Migrate openshift_gcp from shell to deployment manager

### PR DESCRIPTION
**Pull request by pschiffe**
Create infrastructure in GCP with deployment manager instead of shell scripts. Deployment manager configuration is based on https://github.com/openshift/openshift-ansible-contrib/tree/master/reference-architecture/gcp

### CODE REVIEW

**Comment by pschiffe** [comment]
@smarterclayton this is not 100% accurate to the shell scripts, most notably bootstrap scripts are ignored for now, prefix variable doesn't contain dash at the end, some variables may do not have any effect..

Quick question - can I test this PR with `test gcp` and `test launch-gcp`, or that use different code to create GCP infra?

**Comment by pschiffe** [comment]
```
TASK [openshift_gcp : Get ns servers] ******************************************
Friday 09 February 2018  19:03:33 +0000 (0:00:00.858)       0:12:17.227 ******* 
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["gcloud", "--project", "openshift-gce-devel-ci", "dns", "managed-zones", "describe", "origin-ci-int-gce-dev-rhcloud-com", "--format=value(nameServers)"], "delta": "0:00:01.245580", "end": "2018-02-09 19:03:34.702411", "msg": "non-zero return code", "rc": 1, "start": "2018-02-09 19:03:33.456831", "stderr": "ERROR: (gcloud.dns.managed-zones.describe) HTTPError 404: The 'parameters.managedZone' resource named 'origin-ci-int-gce-dev-rhcloud-com' does not exist.", "stderr_lines": ["ERROR: (gcloud.dns.managed-zones.describe) HTTPError 404: The 'parameters.managedZone' resource named 'origin-ci-int-gce-dev-rhcloud-com' does not exist."], "stdout": "", "stdout_lines": []}
```

When using [gcdns_zone](http://docs.ansible.com/ansible/latest/gcdns_zone_module.html) ansible module, it's not possible to specify zone id, it's always `{{ public_hosted_zone | replace('.', '-') }}`. @smarterclayton could be the config for the CI updated to have the `origin-ci-int-gce-dev-rhcloud-com` zone id?

**Comment by smarterclayton** [comment]
It's not clear to me why we would want to update the config if you can always translate it.

File: roles/openshift_gcp/tasks/main.yml [hunk 1]
```diff
@@ -1,43 +1,53 @@
...
-
-- name: Provision GCP DNS domain
-  command: /tmp/openshift_gcp_provision_dns.sh
-  args:
-    chdir: "{{ files_dir }}"
-  register: dns_provision
-  when:
-  - state | default('present') == 'present'
-
-- name: Ensure that DNS resolves to the hosted zone
```
Comment (smarterclayton) [inline roles/openshift_gcp/tasks/main.yml]: Is this lost in your change?  This is the most valuable part for a user.

Comment (pschiffe) [reply roles/openshift_gcp/tasks/main.yml]: No, it was just moved to the `dns-zone.yml` file.

**Comment by pschiffe** [comment]
> It's not clear to me why we would want to update the config if you can always translate it.

@smarterclayton it's not a problem when DNS zone is created with new code, but I guess the zone is already configured with different zone id than `origin-ci-int-gce-dev-rhcloud-com`, so `test gcp` with this PR will always fail. To fix it, `dns_managed_zone` config var in CI must be set to `origin-ci-int-gce-dev-rhcloud-com` as well as the zone id in GCP console. (Is this the current value? https://github.com/openshift/origin-gce/blob/master/playbooks/files/vars.yaml#L20 )

**Comment by pschiffe** [comment]
@smarterclayton ping. We can do a short bluejeans call if you prefer.

**Comment by smarterclayton** [comment]
Going to have to wait until next week.  Also, this is something that has to
land after the static pod move so we don’t delay that.

On Feb 15, 2018, at 8:10 AM, Peter Schiffer <notifications@github.com>
wrote:

@smarterclayton <https://github.com/smarterclayton> ping. We can do a short
bluejeans call if you prefer.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/openshift-ansible/pull/7092#issuecomment-365922533>,
or mute the thread

**Pull request closed without merging**